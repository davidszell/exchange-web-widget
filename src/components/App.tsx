import React from "react";

const App = () => {
    return <h1>Exchange Web Widget</h1>
}

export default App;