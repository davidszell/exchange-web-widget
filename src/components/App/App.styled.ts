import styled from 'styled-components';

export const Container = styled.main`
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
    align-items: center;
    background-color: #157ae2;
    color: white;
`;
