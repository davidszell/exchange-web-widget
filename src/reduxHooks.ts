import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import { AnyAction } from "redux";
import { ThunkDispatch } from "redux-thunk";
import { RootState } from "./store";

export const useAppDispatch = (): ThunkDispatch<RootState, any, AnyAction> => useDispatch<ThunkDispatch<RootState, any, AnyAction>>()
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector