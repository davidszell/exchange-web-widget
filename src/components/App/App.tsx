import React, { FC } from "react";
import { Container } from "./App.styled";

const App: FC = (): JSX.Element => {
    return (
        <Container>
            <div>
                <h1>Sell EUR</h1>
                <p>€1 = $1.16</p>
                <div>
                    <button>EUR</button>
                    <p>Balance: 10</p>
                    <input type="number" placeholder="0"></input>
                </div>
                <button>↓</button>
                <div>
                    <button>USD</button>
                    <p>Balance: 30</p>
                    <input type="number" placeholder="0"></input>
                </div>
                <button>Sell EUR to USD</button>
            </div>
        </Container>
    );
}

export default App;