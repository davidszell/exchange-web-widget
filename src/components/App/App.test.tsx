import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components'
import App from './App';
import * as reduxHooks from '../../reduxHooks';

describe('App', () => {
    const useSelectorMock = jest.spyOn(reduxHooks, 'useAppSelector')
    const useDispatchMock = jest.spyOn(reduxHooks, 'useAppDispatch')

    beforeEach(() => {
        useSelectorMock.mockImplementation(selector => selector(mockStore));
        useDispatchMock.mockReturnValue(jest.fn())
    })
    afterEach(() => {
        useSelectorMock.mockClear()
        useDispatchMock.mockClear()
    })

    const mockStore = {
        wallets: [
            {
                name: "EUR",
                balance: 10,
                longName: "Euros",
                symbol: "€"
            },
            {
                name: "GBP",
                balance: 20,
                longName: "British Pounds",
                symbol: "£"
            },
            {
                name: "USD",
                balance: 30,
                longName: "US Dollars",
                symbol: "$"
            }
        ],
        exchangeRates: {
            base: "EUR",
            rates: {
                "EUR": 1,
                "GBP": 0.85,
                "USD": 1.16
            }
        }
    };
    it('matches snapshot', () => {
        const tree = renderer
            .create(<App />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})