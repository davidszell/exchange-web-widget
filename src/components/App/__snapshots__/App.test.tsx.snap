// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<main
  className="c0"
>
  <div
    className=""
  >
    <h1
      data-testid="header"
    >
      Sell
       
      EUR
    </h1>
    <p
      data-testid="exchangeRate"
    >
      €
      1 = 
      £
      0.85
    </p>
    <div>
      <button
        onClick={[Function]}
      >
        EUR
      </button>
      <p>
        Balance: 
        10
      </p>
      <input
        onChange={[Function]}
        placeholder="0"
        type="number"
        value={0}
      />
    </div>
    <button
      data-testid="directionButton"
      onClick={[Function]}
    >
      ↓
    </button>
    <div>
      <button
        onClick={[Function]}
      >
        GBP
      </button>
      <p>
        Balance: 
        20
      </p>
      <input
        onChange={[Function]}
        placeholder="0"
        type="number"
        value={0}
      />
    </div>
    <button
      data-testid="exchangeButton"
    >
      Sell
       
      EUR
       
      for
       
      GBP
    </button>
  </div>
</main>
`;
